### PR TITLE
[sort cmd] add numeric sorting

### LINKED ISSUE #7743: `:sort n` sorts lines lexicographically instead of numerically
**Describe the bug**

Regular Vim has the `:sort` command for sorting lines. By default `:sort` goes with lexicographic order, as in `a < aa < b`. With numbers this looks like this: `1 < 11 < 2`. But that's not numeric order, as in `1 < 2 < 11`. So regular Vim also has `:sort n` for numeric ordering. In VSCodeVim however, `:sort n` goes with lexicographic order. 

Incidentally, there is also `:sort u` if you want to sort lexicographically but also remove duplicates. This works. Just to prove that `:sort` doesn't just ignore its arguments.

**To Reproduce**

Select a list of numbers in visual mode. For example:

```
1
10
2
2
```

and type `:sort n`.

**Expected behavior**

The lines are sorted numerically. In case of the above example:

```
1
2
2
10
```

**Screenshots**

https://user-images.githubusercontent.com/26570572/172331276-52613b02-844a-4e6c-a1e0-869a0dfe49a3.mov

Regular Vim in comparison:

https://user-images.githubusercontent.com/26570572/172332801-3e395c45-1dea-4c30-9b23-b102a0766fc5.mov

**Environment (please complete the following information):**

- Issue Type: `Bug`
- Extension Name: `vim`
- Extension Version: `1.22.2`
- OS Version: `Darwin x64 21.4.0`
- VS Code version: `1.66.2`

```json 
{
	"messages": [],
	"activationTimes": {
		"codeLoadingTime": 156,
		"activateCallTime": 2,
		"activateResolvedTime": 309,
		"activationReason": {
			"startup": true,
			"extensionId": {
				"value": "vscodevim.vim",
				"_lower": "vscodevim.vim"
			},
			"activationEvent": "*"
		}
	},
	"runtimeErrors": [],
	"runningLocation": {
		"affinity": 0,
		"kind": 1
	}
}
```

### PR DESCRIPTION
**Pull request by jan25**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**

Fixes #7743

**Special notes for your reviewer**:

I couldn't get tests running locally, I'll wait for CI for it.

I extended `NumericString` module and it can be used in more sorting options - e.g. oct, hex sorting in future.

### CODE REVIEW

**Comment by J-Fields** [comment]
@jan25 I'm interested in this change - would you mind fixing the build please?

File: test/number/numericString.test.ts [hunk 1]
```diff
@@ -30,4 +30,18 @@ suite('numeric string', () => {
...
+    assert.strictEqual(input, NumericString.parse(input)?.num.toString());
+    assert.strictEqual(input, NumericString.parse(input)?.num.toString());
+  });
+
+  test('handles decimal radix', () => {
+    assert.strictEqual(NumericString.parse('07', NumericStringRadix.Dec)?.num.value, 7);
+    assert.strictEqual(NumericString.parse('07', NumericStringRadix.Dec)?.num.negative, false);
+
+    assert.strictEqual(NumericString.parse('hi-07hello', NumericStringRadix.Dec)?.num.value, -7);
+    assert.strictEqual(NumericString.parse('hi-07hello', NumericStringRadix.Dec)?.num.negative, true);
```
Comment (jan25) [inline test/number/numericString.test.ts:45]: this covers the bug fix for negative flag.

File: src/common/number/numericString.ts [hunk 2]
```diff
@@ -98,15 +112,13 @@ export class NumericString {
     // remove the width of negative sign.
     const numLength = coreLength - NumericString.numPrefix[coreRadix].length - (coreSign ? 1 : 0);
 
+    const negative = coreSign;
```
Comment (jan25) [inline src/common/number/numericString.ts:115]: This fixes bug with negative flag because coreSign indicates match has `-` prefix,

**Comment by jan25** [comment]
Sadly i couldn't get tests running locally. Let's see if build passes.

**Review by J-Fields — APPROVED** [review]
Thanks!

**Pull request merged**